templates/search/blocks/search_pagination.html

```diff
@@ -1,53 +1,46 @@
 {% load search_tags humanize %}
-
-<nav class="pagination" role="navigation" aria-label="Records pagination" data-container-name="Pagination"
-     id="analytics-disambiguation-pagination">
-    {% if page.has_other_pages %}
-    <ul class="pagination__list">
-        {% if page.has_previous %}
-        <li class="pagination__list-item">
-
-          <a class="pagination__page-chevron-previous" href="?{% query_string_include 'page' page.previous_page_number %}" data-link-type="Pagination" aria-label="Go to Previous Page" data-link="Previous page">&lt; Previous</a>
-
-        </li>
-        {% else %}
-        <li class="pagination__list-item">
-            <span class="pagination__page-chevron-previous--disabled">&lt; Previous</span>
-        </li>
-        {% endif %}
-        <li class="pagination__list-pages" aria-label="Page numbers">
-            <ol>
-                {% for i in page_range %}
-                <li class="pagination__list-item">
-                    {% if page.number == i %}
-                    <a class="pagination__page-link-current" href="?{% query_string_include 'page' i %}" aria-label="Current page, Page {{ i }}" aria-current="true" data-link-type="Pagination" data-link="{{ i }}">
-                        {{ i }}
-                    </a>
-                    {% else %}
-                    {% if i == page.paginator.ELLIPSIS %}
-                    {{ i }}
-                    {% else %}
-                    <a class="pagination__page-link" href="?{% query_string_include 'page' i %}" data-link-type="Pagination" aria-label="Go to page {{ i }}" data-link="{{ i }}">
+{% if page.has_other_pages %}
+<nav class="tna-pagination" role="navigation" aria-label="Pagination" id="analytics-disambiguation-pagination">
+    {% if page.has_previous %}
+    <div class="tna-pagination__prev">
+      <a class="tna-button tna-pagination__link" href="?{% query_string_include 'page' page.previous_page_number %}" role="button" rel="prev" data-link-type="Pagination"  data-link="Previous page">
+        <svg class="tna-pagination__icon tna-pagination__icon--prev" xmlns="http://www.w3.org/2000/svg" height="13" width="15" aria-hidden="true" focusable="false" viewBox="0 0 15 13">
+          <path d="m6.5938-0.0078125-6.7266 6.7266 6.7441 6.4062 1.377-1.449-4.1856-3.9768h12.896v-2h-12.984l4.2931-4.293-1.414-1.414z"></path>
+        </svg>
+        <span class="tna-pagination__link-title">
+          Previous
+        </span>
+      </a>
+    </div>
+    {% endif %}
+    <ul class="tna-pagination__list">
+        {% for i in page_range %}
+            {% if i == page.paginator.ELLIPSIS %}
+            <li class="tna-pagination__item tna-pagination__item--ellipses">&ctdot;</li>
+            {% else %}
+            {# API will return up to 5020 results so cap the pagination there #}
+            {% if i < 252 %}
```

Comment:
Nice one. 
- Should we have this limit as a BE attribute for the template?
- Is there a UX need/implication - to let the user know about the limit?
- When it crosses the limit (Internal error). What is appropriate in this case?
```
Response body: {
    "timestamp": "2023-11-22T09:04:41.890",
    "status": "Bad Request",
    "statusCode": 400,
    "message": "search.from: This service is not intended for deep pagination",
    "path": "/api/v1/data/search"
}
```

Reply:
That's a good point. On ETNA when this happens, the user is just presented with a 500 page https://beta.nationalarchives.gov.uk/search/catalogue/?group=tna&page=252. Handling this on the BE might be a more elegant solution, as you suggest. I believe this restriction is a standard limitation set by elastic search. On other projects I've worked on with K-Int, this restriction has been increased to support the full dataset. So this could be removed in that case.

As we're currently working with a different dataset and not the OHOS data, do you know the number of records we'll be expecting to see when this is using real data?

Reply:
Raised it in slack